pkg/operator/ceph/csi/template/rbd/csi-rbdplugin-provisioner-dep.yaml

```diff
@@ -268,10 +298,56 @@ spec:
               mountPath: /csi
           imagePullPolicy: {{ .ImagePullPolicy }}
         {{ end }}
+        {{ if .CSILogRotation }}
+        - args:
+          - |
+            echo "Starting the csi-logrotate-sidecar"
+            mkdir -p /logrotate-config;
+            echo '/var/lib/cephcsi/log/*.log {
+                daily
+                missingok
+                rotate 7
+                compress
+                copytruncate
+                notifempty
+            }' > /logrotate-config/csi;
+            echo "File creation container completed";
+            LOG_ROTATE_CEPH_FILE=/logrotate-config/csi
+            PERIODICITY={{ .CSILogRotationPeriod }}
+            if [ "$PERIODICITY" != "" ]; then
+              echo "replace default daily with given user input"
+              sed --in-place "s/daily/$PERIODICITY/g" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            LOG_MAX_SIZE={{ .CSILogRotationMaxSize }}
+            if [ "$LOG_MAX_SIZE" != "0" ]; then
+              sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            while true; do
+              logrotate --verbose "$LOG_ROTATE_CEPH_FILE"
+              sleep 15m
+            done
```

Comment:
can we do something like below which will avoid string replacement
```yaml
        env:
        - name: LOG_ROTATE_FREQUENCY
          value: "daily"
        - name: LOG_ROTATE_MAXFILES
          value: "5"
        command: ["/bin/sh", "-c"]
        args:
          - |
            echo "/var/log/cso/*.log {
                su root root
                rotate ${LOG_ROTATE_MAXFILES}
                ${LOG_ROTATE_FREQUENCY}
                compress
                delaycompress
                copytruncate
                missingok
                notifempty
            }" > /etc/logrotate.conf && \
            while true; do logrotate -f /etc/logrotate.conf; sleep 3600; done
```

Reply:
I have to only add maxsize if it is there, so for that still used update file method using sed